core/storage/question/gae_models.py

```diff
@@ -260,6 +260,92 @@ def get_question_skill_links_by_skill_ids(
         )
         return question_skill_link_models, next_cursor_str
 
+    @classmethod
+    def get_question_skill_links_based_on_mastery_equidistributed_by_skill(
+            cls, total_question_count, skill_ids, degrees_of_mastery):
+        """Fetches the list of constant number of QuestionSkillLinkModels
+        linked to the skills, sorted by the difference between skill
+        difficulty and skill mastery.
+
+        Args:
+            total_question_count: int. The number of questions expected.
+            skill_ids: list(str). The ids of skills for which the linked
+                question ids are to be retrieved.
+            degrees_of_mastery: dict(str, float). The keys are the requested
```

Comment:
From the implementation, looks like it could be None too, right? Please explain this case in the docstring. Make sure your docstrings are carefully written.

Reply:
Done.